feat(linter): implement `eslint/no-invalid-regexp`

Boshen opened the pull request:
closes #611

@leaysgur Kicking of `no-invalid-regexp`, feel free to take over and claim the bounty on #611 :grin: 

I can continue if you wanna work on other stuff (getting confused in prettier land ;-))

Boshen commented on crates/oxc_linter/src/rules/eslint/no_invalid_regexp.rs:
```diff
@@ -0,0 +1,240 @@
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
```
Some of these may have incorrect `\`, please double check with https://github.com/eslint/eslint/blob/main/tests/lib/rules/no-invalid-regexp.js

leaysgur commented:
OK, I will take this. 🤭 (Prettier... too far to go...)

leaysgur commented:
Basically done.

But I'd like to fix diagnostic labels in the parser to align display.

leaysgur commented:
Benchmark...why... 😮

leaysgur commented:
...🤷🏻‍♂️ 

![image](https://github.com/user-attachments/assets/e748a326-1d1e-4e3a-bfe8-3f23fe67a867)

Boshen commented:
Strange ... I'll take a look.

leaysgur commented:
According to https://github.com/oxc-project/oxc/pull/5570#issuecomment-2335165856 , is it because `New|CallExpression` is called as frequently as `UpdateExpression`?

But the commit just before the performance problem was 1e26fce .

![image](https://github.com/user-attachments/assets/4ef13e27-fa01-40ce-81bd-a7b10e8c6900)

And here is a diff of that commit and the latest.

https://github.com/oxc-project/oxc/compare/1e26fce...6a8c379

Since I only adjusted `Span`, why were there no performance issues at this point...?

Boshen commented:
Codspeed is drunk, I don't see anything wrong.

Boshen committed:
feat(linter): implement `eslint/no-invalid-regexp` (#5443)

closes #611

@leaysgur Kicking of `no-invalid-regexp`, feel free to take over and claim the bounty on #611 :grin:

I can continue if you wanna work on other stuff (getting confused in prettier land ;-))
